touchforms/backend/xformplayer.py

```diff
@@ -40,77 +41,52 @@
 class NoSuchSession(Exception):
     pass
 
-class global_state_mgr(object):
+class GlobalStateManager(object):
     instances = {}
     instance_id_counter = 0
 
-    session_cache = {}
-    
     def __init__(self, ctx):
         self.ctx = ctx
         self.lock = threading.Lock()
         self.ctx.setNumSessions(0)
-    
-    def new_session(self, xfsess):
+        self.redis = redis.StrictRedis(
+            host=settings.REDIS_HOST,
+            port=settings.REDIS_PORT,
+            db=settings.REDIS_DB,
+        )
+
+    def new_session(self, xform_session):
         with self.lock:
-            self.session_cache[xfsess.uuid] = xfsess
-            self.ctx.setNumSessions(len(self.session_cache))
+            state = xform_session.session_state()
+            self.redis.set(xform_session.uuid, json.dumps(state))
+            #self.ctx.setNumSessions(len(self.session_cache))
```

Comment:
Not sure what to do about this guy. Seems like this is only used for the offline mode. I don't think I've ever seen the offline mode